File: lib/utilities.ex
defmodule Utilities do
  def lines(filename) do
    File.open!(filename, fn pid ->
      pid
      |> IO.stream(:line)
      |> Enum.map(&String.trim/1)
      |> Enum.into([])
    end)
  end

  def lines(filename, parse_action) do
    File.open!(filename, fn pid ->
      pid
      |> IO.stream(:line)
      |> Enum.map(&String.trim/1)
      |> Enum.map(parse_action)
      |> Enum.into([])
    end)
  end
end

File: clojure/src/utilities.clj
(ns utilities
  (:require [clojure.java.io :as io]))

(defn read-file [filename]
  (with-open [reader (io/reader filename)]
    (->> reader
         (line-seq)
         (reduce conj []))))

(defn get-lines [[filename & _rest]]
  (->> filename
       (read-file)))

(def pattern #"#(?<id>\d++)[^\d]++(?<x>\d++)[^\d]++(?<y>\d++)[^\d]++(?<width>\d++)[^\d]++(?<height>\d++)")

(defn matcher [line]
  (re-matcher pattern line))
